modeling/game_ends_and_starts_again.plantuml

```diff
@@ -0,0 +1,52 @@
+@startuml
+
+title Game ends and starts again with 2 players
+
+note right 
+Game is over when:
+- I don't have any cards
+- No cards left in Ocean
+- No one else has any cards left
+end note
+
+
+note right 
+Current state:
+John has [3 of spades, 3 of diamonds]
+Simon has [3 of clubs]
+Ocean has [3 of hearts]
+end note
+
+John -> Simon: give_me_all_your(3)
+Simon --> John: [3 of clubs]
+Simon -> Controller: out_of_cards([4,5])
+John -> Simon: give_me_all_your(3)
+Simon --> John: go_fish
+John -> Ocean: take_card()
+Ocean --> John: [3 of hearts]
+Ocean -> Controller: out_of_cards()
+John -> John: save_book
+note right 
+John got a book and now doesn't have any cards.
+Simon doesn't have any cards.
+The Ocean is empty.
+game is over
+end note
+
+John -> Controller: out_of_cards([3])
+Controller -> Controller: game_over
+Controller -> John: stop()
+Controller -> Simon: stop()
+Controller -> Ocean: stop()
+Ocean -> Controller: :ok
+
+
+note right
+    Controller sets winner in its state
+    Controller has now set the :game_state to :game_over
+end note
+
+John -> Controller: start_game()
+
+
+@enduml
```

Comment:
What information does this diagram convey, that isn't conveyed in the new sequence diagram?

If none then we should delete this one.

If some then we should destil it down to focus on showing only that.

Reply:
Yeah this is a leftover that can be deleted I think